tests/test_job.py

```diff
@@ -943,7 +972,90 @@ def test_dependents_are_met_if_dependency_is_deleted(self):
 
         assert dependent_job.dependencies_are_met()
         assert dependent_job.get_status() == JobStatus.QUEUED
-    
+
+    def test_dependencies_are_met_at_execution_time(self):
+        queue = Queue(connection=self.testconn)
+
+        queue.enqueue_call(fixtures.say_hello, job_id="A")
+        queue.enqueue_call(fixtures.say_hello, job_id="B")
+        job_C = queue.enqueue_call(fixtures.check_dependencies_are_met, job_id="C", depends_on=["A", "B"])
```

Comment:
Nitpick: can we use the regular `queue.enqueue()` API here? Tests will be cleaner this way, I think `enqueue_call` should be used only when we're specifically testing it.